lib/bundler/cli/outdated.rb

```diff
@@ -162,17 +153,13 @@ def run
   private
 
     def retrieve_active_spec(strict, definition, current_spec)
-      if strict
-        active_spec = definition.find_resolved_spec(current_spec)
-      else
-        active_specs = definition.find_indexed_specs(current_spec)
-        if !current_spec.version.prerelease? && !options[:pre] && active_specs.size > 1
-          active_specs.delete_if {|b| b.respond_to?(:version) && b.version.prerelease? }
-        end
-        active_spec = active_specs.last
-      end
+      return definition.find_resolved_spec(current_spec) if strict
 
-      active_spec
+      active_specs = definition.find_indexed_specs(current_spec)
+      if !current_spec.version.prerelease? && !options[:pre] && active_specs.size > 1
+        active_specs.delete_if {|b| b.respond_to?(:version) && b.version.prerelease? }
+      end
+      active_specs.last
```

Comment:
Regarding this method, I can see how removing the `active_spec` local makes it simpler, but in general I don't think early returns always make the code better. In this case, I prefer the old version because it gives both branches the same importance.

Reply:
do you mean rollback of `retrieve_active_spec` method?

Reply:
Yeah